app/controllers/todos_controller.rb

```diff
@@ -0,0 +1,59 @@
+class TodosController < ApplicationController
+  before_action :set_todo_list
+  before_action :set_todo, only: [:toggle_status, :update, :destroy]
+
+  # GET /users/:user_id/todo_lists/:todo_list_id/todos
+  def index
+    @todos = @todo_list.todos
+    render json: @todos
+  end
+
+  # POST /users/:user_id/todo_lists/:todo_list_id/todos
+  def create
+    @todo = @todo_list.todos.build(todo_params)
+    if @todo.save
+      render json: @todo, status: :created
+    else
+      render json: { error: @todo.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH /users/:user_id/todo_lists/:todo_list_id/todos/:id/toggle_status
+  def toggle_status
+    @todo.toggle_status!
+    render json: @todo
+  rescue ActiveRecord::RecordInvalid => e
+    render json: { error: e.message }, status: :unprocessable_entity
+  end
```

Comment:
What's the usecase of this endpoint?

Reply:
like basically if we want to change the state from pending to complete through the frontend perspective: like: User clicks "Mark as Complete" on the frontend.
then a request is sent to the toggle_status endpoint.
and if todos wasnt able to update bcz of validation error then it will show case error msg.
so i tried to assume with respect to frontend view so i thought for this end point.